lib/extensions/database_cleaner-activerecord-seeded_deletion.rb

```diff
@@ -1,12 +1,18 @@
 require 'database_cleaner-active_record'
+require 'thread'
 
 module DatabaseCleaner
   module ActiveRecord
     # SeededDeletion is a strategy that deletes all records from tables except those that existed before it was instantiated
     # This is useful for tests that need the seeded data to be present.
     class SeededDeletion < Deletion
+      # Class level mutex for thread safety around start/clean actions.
+      @mutex = Mutex.new
+
       def clean
-        connection.transaction(:requires_new => true) do
+        # Use a transaction with serializable isolation to prevent other threads
+        # from modifying the tables during deletion
+        connection.transaction(:requires_new => true, :isolation => :read_committed) do
```

Comment:
Switched to read_committed... it seems to prevent the errors we were seeing locally.